wrapper/wrapper.rs

```diff
@@ -135,3 +146,98 @@ pub unsafe extern "C" fn descriptor_address_at(
         })),
     }
 }
+
+struct TestType(types::Type);
+
+impl fmt::Display for TestType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self.0.corr.base {
+            types::Base::B => "B",
+            types::Base::K => "K",
+            types::Base::V => "V",
+            types::Base::W => "W",
+        })?;
+        f.write_str(match self.0.corr.input {
+            types::Input::Zero => "z",
+            types::Input::One => "o",
+            types::Input::OneNonZero => "on",
+            types::Input::Any => "",
+            types::Input::AnyNonZero => "n",
+        })?;
+        if self.0.corr.dissatisfiable {
+            fmt::Write::write_char(f, 'd')?;
+        }
+        if self.0.corr.unit {
+            fmt::Write::write_char(f, 'u')?;
+        }
+        f.write_str(match self.0.mall.dissat {
+            types::Dissat::None => "f",
+            types::Dissat::Unique => "e",
+            types::Dissat::Unknown => "",
+        })?;
+        if self.0.mall.safe {
+            fmt::Write::write_char(f, 's')?;
+        }
+        if self.0.mall.non_malleable {
+            fmt::Write::write_char(f, 'm')?;
+        }
+        Ok(())
+    }
+}
+
+pub struct MiniscriptProperties {
+    types: String,
+    op_codes: usize,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn miniscript_parse(ptr: u32, len: u32) -> u64 {
+    let result = || -> Result<MiniscriptProperties, String> {
+        let miniscript_string = ptr_to_string(ptr, len);
+        let ms =
+            miniscript::Miniscript::<String, miniscript::Segwitv0>::from_str_insane(&miniscript_string)
+                .map_err(|e| e.to_string())?;
+                
+        let ms_types = format!("{}", TestType(ms.ty));
+        let mut ms_types = ms_types.chars().collect::<Vec<_>>();
+        ms_types.sort_by(|a, b| b.cmp(a));
+
+        Ok(MiniscriptProperties{
+            types: ms_types.into_iter().collect(),
+            op_codes: ms.ext.ops.count,
+        })
+    };
+    match result() {
+        Ok(props) => json_to_ptr(serde_json::json!({
+            "types": props.types,
+            "op_codes": props.op_codes,
+        })),
+        Err(err) => json_to_ptr(serde_json::json!({
+            "error": err,
+        })),
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn miniscript_compile(ptr: u32, len: u32) -> u64 {
```

Comment:
Same, please add a comment explaining the input and output, and that it is currently only used in unit tests.